### PR TITLE
initial test for placeholders

### PR DESCRIPTION
**Pull request by abilandz**

### CODE REVIEW

File: PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h [hunk 1]
```diff
@@ -499,6 +558,150 @@ void BookWeightsHistograms()
...
+        }
+        else if(ETAKINE == v  && fUseCustomKineDependenceBins[ETAq])
+        {
+         fNestedLoopsPro[k][n][v] = new TProfile(Form("fNestedLoopsPro[%d][%d][%d]",k,n,v),Form("#LT#LTcos[%s(%s)]#GT#GT",1==n+1?"":Form("%d",n+1),oVariable[k].Data()),fKineDependenceBins[ETAq]->GetSize()-1,fKineDependenceBins[ETAq]->GetArray());
+        }
+        else
+        {
+        */
+        // the default binning:
+        // fNestedLoopsPro[k][n][v] = new TProfile(Form("fNestedLoopsPro[%d][%d][%d]",k,n,v),Form("#LT#LTcos[%s(%s)]#GT#GT",1==n+1?"":Form("%d",n+1),oVariable[k].Data()),vvvariableNBins[v],vvvariableMinMax[v][0],vvvariableMinMax[v][1]);
```
Comment (victor-gonzalez) [inline PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h:638]: Please don't leave code commented

File: PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h [hunk 2]
```diff
@@ -499,6 +558,150 @@ void BookWeightsHistograms()
...
+        nl_a.fNestedLoopsPro[k][n][v]->GetXaxis()->SetTitle(vvVariable[v].Data());
+        // fNestedLoopsPro[k][n][v]->SetFillColor(colorsW[v]-10);
+        // fNestedLoopsPro[k][n][v]->SetLineColor(colorsW[v]);
+        /*
+        if(fUseFixedNumberOfRandomlySelectedParticles && 1==v) // just a warning for the meaning of multiplicity in this special case
+        {
+         nl_a.fNestedLoopsPro[k][n][1]->GetXaxis()->SetTitle("WARNING: for each multiplicity, fFixedNumberOfRandomlySelectedParticles is selected randomly in Q-vector");
+        }
+        */
+
```
Comment (victor-gonzalez) [inline PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h:652]: Same comment

File: PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h [hunk 3]
```diff
@@ -499,6 +558,150 @@ void BookWeightsHistograms()
...
 
+void BookNestedLoopsHistograms()
+{
+  // Book all nested loops histograms.
+
+  // a) Book the profile holding flags;
+  // *) ...
+
+  if (fVerbose) {
+    Green(__PRETTY_FUNCTION__);
```
Comment (victor-gonzalez) [inline PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h:569]: I don't know what `Green` does but, please, for logging you have to use the O2 logging machinery

File: PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h [hunk 4]
```diff
@@ -727,6 +939,123 @@ void CalculateCorrelations()
...
+   }
+  }
+   cout<<"nParticles = "<<nParticles<<endl;
+  */
+
+  // 2p:
+  if (nParticles < 2) {
+    return;
+  }
+  cout << "      CalculateNestedLoops(void), 2-p correlations .... " << endl;
```
Comment (victor-gonzalez) [inline PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h:969]: Don't use `cout`, use O2 logging machinery

File: PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h [hunk 5]
```diff
@@ -727,6 +939,123 @@ void CalculateCorrelations()
...
+  nBinsQV = c_a.fCorrelationsPro[0][0][0]->GetNbinsX();
+  nBinsNL = nl_a.fNestedLoopsPro[0][0][0]->GetNbinsX();
+  if (nBinsQV != nBinsNL) {
+    cout << __LINE__ << endl;
+    exit(1);
+  }
+  cout << endl;
+  cout << "   [0] : integrated" << endl;
+  for (Int_t o = 0; o < 4; o++) {
+    cout << Form("   ==== <<%d>>-particle correlations ====", 2 * (o + 1)) << endl;
```
Comment (victor-gonzalez) [inline PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h:1030]: Same comment

File: PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h [hunk 6]
```diff
@@ -727,6 +939,123 @@ void CalculateCorrelations()
...
+      for (Int_t b = 1; b <= nBinsQV; b++) {
+        if (c_a.fCorrelationsPro[o][h][0]) {
+          valueQV = c_a.fCorrelationsPro[o][h][0]->GetBinContent(b);
+        }
+        if (nl_a.fNestedLoopsPro[o][h][0]) {
+          valueNL = nl_a.fNestedLoopsPro[o][h][0]->GetBinContent(b);
+        }
+        if (TMath::Abs(valueQV) > 0. && TMath::Abs(valueNL) > 0.) {
+          cout << Form("   h=%d, Q-vectors:    ", h + 1) << valueQV << endl;
+          cout << Form("   h=%d, Nested loops: ", h + 1) << valueNL << endl;
```
Comment (victor-gonzalez) [inline PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h:1041]: Same comment

File: PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h [hunk 7]
```diff
@@ -727,6 +939,123 @@ void CalculateCorrelations()
...
+          if (TMath::Abs(valueQV - valueNL) > 1.e-5) {
+            cout << Form("[%d][%d][%d]", o, h, 0) << endl;
+            cout << __LINE__ << endl;
```
Comment (victor-gonzalez) [inline PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h:1044]: Same comment

File: PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h [hunk 8]
```diff
@@ -885,38 +1237,51 @@ TH1D* GetHistogramWithWeights(const char* filePath, const char* variable)
...
+    // Check if the external ROOT file exists at specified path:
+    if (gSystem->AccessPathName(filePath, kFileExists)) {
+      Red(Form("if(gSystem->AccessPathName(filePath,kFileExists)), filePath = %s", filePath));
+      cout << __LINE__ << endl;
+      exit(1);
+    }
+    weightsFile = TFile::Open(filePath, "READ");
+  } // if(bFileIsInAliEn)
+
+  // f) Access the external ROOT file and fetch the desired histogram with weights:
```
Comment (victor-gonzalez) [inline PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h:1269]: I don't think it is intended to open general `alien` access from the grid nodes
At least for corrections files you should consider using CCDB

File: PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h [hunk 9]
```diff
@@ -925,12 +1290,171 @@ TH1D* GetHistogramWithWeights(const char* filePath, const char* variable)
...
+  } else {
+    // d) Handle the local case:
+    // Check if the external ROOT file exists at specified path:
+    if (gSystem->AccessPathName(filePath, kFileExists)) {
+      Red(Form("if(gSystem->AccessPathName(filePath,kFileExists)), filePath = %s", filePath));
+      cout << __LINE__ << endl;
+      exit(1);
+    }
+    oaFile = TFile::Open(filePath, "READ");
+  } // if(bFileIsInAliEn)
```
Comment (victor-gonzalez) [inline PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h:1345]: Same comment as above

File: PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h (hunk 3 above)
Comment (abilandz) [reply PWGCF/MultiparticleCorrelations/Core/MuPa-MemberFunctions.h:569]: "Green" is a simple utility function, which prints its arguments in green, this is how it's implemented:

`void Green(const char* text)
{
  cout << "\n\033[1;32m" << text << "\033[0m\n"
       << endl;
}`

It uses the standard escape sequences for colors(*), recognized by other languages as well. This gives me some uniformity when I have to implement the same thing across different languages. 

(*) 
`Black        0;30     Dark Gray     1;30
Red          0;31     Light Red     1;31
Green        0;32     Light Green   1;32
Brown/Orange 0;33     Yellow        1;33
Blue         0;34     Light Blue    1;34
Purple       0;35     Light Purple  1;35
Cyan         0;36     Light Cyan    1;36
Light Gray   0;37     White         1;37`

**Pull request merged**